src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -423,17 +374,100 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, HTTPVer
             }
         }
 
+        private void CreateEntryComponent(HttpContext httpContext, EntryComponent entryComponent)
```

Comment:
I think this method should return an EntryComponent instead of taking it as an input argument. The current use case for it is passing in an unmodified new EntryComponent. I think it makes more sense for the method to make the new EntryComponent itself and pass it back.